protocol/protobuf/proto/state_representation/state.proto

```diff
@@ -13,6 +13,8 @@ enum StateType {
   TRAJECTORY = 5;
   GEOMETRY_SHAPE = 6;
   GEOMETRY_ELLIPSOID = 7;
+  PARAMETER_INT = 21;
```

Comment:
See the comment above this enum:
>The values and order of this enumeration must match the original state_representation::StateType in C++.

This is because the clproto binding casts directly between proto::StateType and state_representation::StateType

```c++
proto::StateType encoder(const StateType& type) {
  return static_cast<proto::StateType>(type);
}
```

In this case, you have extended the enum values of the state_rep::StateType, causing the values of each subsequent field to increase. 21 / 22 would actually map to matrix and vector. While we shouldn't change or replace the field numbers of messages without cause, the values of enums are a bit different (since they are values more than they are field encodings), so you have two options.


You should renumber the enums here to be sequential, matching the new StateType layout, or you should put the new state type at the end of the enum to keep things sequential. I think ultimately the second option is more safe, despite it not looking as nice - but the visuals are a lesser concern.